dom/src/main/scala/fs2/dom/EventTargetHelpers.scala

```diff
@@ -39,9 +39,9 @@ private[dom] object EventTargetHelpers {
           F.delay {
             target.addEventListener[E](
               `type`,
-              (ev: E) => dispatcher.unsafeRunAndForget(ch.send(ev)),
-              new EventListenerOptions {
-                signal = abort
+              (ev: E) => dispatcher.unsafeRunAndForget(ch.send(ev)), {
+                options.signal = abort
```

Comment:
Hm, maybe we should only set that if caller didn't set it already

Reply:
Hm, I didn't think about this. I don't think the user should be setting this: if they want to cancel the event, then they should cancel the stream. So I think it's okay to override.

Of course, this suggests again that maybe we should have our own `EventListenerOptions` to disallow this. But still I would prefer to avoid that if possible.

Reply:
We certainly can, but overriding caller intent can be confusing. Without overriding `signal` this function would be pretty left-unique, meaning it would "do" something different for each different set of params. Also, I admit, I dislike that mutable Options thing :smile: What about a plain class to keep bincompat issues out of the door?

Reply:
> Without overriding `signal` this function would be pretty left-unique, meaning it would "do" something different for each different set of params.

Sorry, I'm not really sure what you mean :)

> What about a plain class to keep bincompat issues out of the door?

That would be fine for bincompat. The reasons I prefer the mutable options thingy are:
1. it has a simple user API for selecting which fields they want to set
2. less code for us to maintain, but at this point admittedly this is a lost cause 😆 
3. less allocations and inefficiency to convert it to the equivalent JS type

Mostly I care about (1) I guess. If you want maybe open a separate PR to `main` for that? Since it affects existing code.

Reply:
> Sorry, I'm not really sure what you mean :)

I meant `listen(a, b, new EventListenerOptions {signal = c}) = listen(a, b, new EventListenerOptions {signal = d} = listen(a, b, new EventListenerOptions {signal = e} = ...)`

Ok, I'll make a speparate PR for this topic.